src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java

```diff
@@ -29,34 +29,55 @@ public ClientServiceImpl(BookRepository bookRepository, ClientRepository clientR
     public Book borrowBook(Long bookId, Long clientId) {
         Optional<Client> optionalClient = clientRepository.findById(clientId);
         Optional<Book> optionalBook = bookRepository.findById(bookId);
-        if (optionalClient.isPresent() && optionalBook.isPresent()) {
+        //added borrow validation
+        if (optionalClient.isPresent() && optionalBook.isPresent()&&optionalClient.get().getBooks().size()<2 ){
```

Comment:
What if validation is not met, what is propagated back to client?

Reply:
expection added